### PR TITLE
FET-690 Workspace settings backend decorator

### PR DESCRIPTION
**Pull request by tomassatinsky**
---

Supported PR commands:

| Command                | Description            |
| ---------------------- | ---------------------- |
| `ok to test`           | Re-run standard checks |
| `extended test`        | BackstopJS tests       |
| `extended check sonar` | SonarQube tests        |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/4003/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5640/ : SUCCESS in 17m 02s

File: libs/sdk-backend-base/src/workspaceSettingsBackend/index.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+/**
+ * Specifies workspace settings and current user settings to be adjusted by the decorator
+ *
+ * @beta
+ */
+export interface WorkspaceSettingsConfiguration {
+    settingsWrapper?: SettingsWrapper;
+    currentUserSettingsWrapper?: CurrentUserSettingsWrapper;
+}
```
Comment (lupko) [inline libs/sdk-backend-base/src/workspaceSettingsBackend/index.ts:63]: please describe the settings vs current user settings wrapper & the default behavior.

Comment (lupko) [reply libs/sdk-backend-base/src/workspaceSettingsBackend/index.ts:63]: also, I propose to introduce the convenience where it is possible to specify settings for both workspace settings & workspace-user settings in a single place.

the idea is... often time, you are interested in one set of ISettings to be applied in any context (user-workspace, workspace etc). with the current setup, you would have to specify these settings in via two wrappers.

the proposal is, have a single setting that can influence both.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/4011/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5648/ : SUCCESS in 15m 43s

File: libs/sdk-backend-base/src/workspaceSettingsBackend/index.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+/**
+ * Specifies workspace settings and current user settings to be adjusted by the decorator
+ *
+ * @beta
+ */
+export interface WorkspaceSettingsConfiguration {
+    settingsWrapper?: SettingsWrapper;
+    currentUserSettingsWrapper?: CurrentUserSettingsWrapper;
+    commonSettingsSetter?: CommonSettingsSetter;
```
Comment (lupko) [inline libs/sdk-backend-base/src/workspaceSettingsBackend/index.ts:80]: can you please add docs on how/when are these three used?

**Review by lupko — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result
 - No issue found. Congratulations!

More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1443

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Pull request merged**